Named tensor support for: all, any, bitwise_not, cumprod, cumsum, and more

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26829 Fix CUDA named tensor `copy_`
* **#26815 Named tensor support for: all, any, bitwise_not, cumprod, cumsum, and more**
* #26563 Named tensor support for logsumexp, mode, kthvalue, median, min, max

This PR adds named tensor support for:
- any, all, `bitwise_not(_)`, cumprod, cumsum, `logical_not`

In addition, it adds smoke tests for a variety of tensor attributes and
fns:
- is_shared, is_signed
- retain_grad, register_hook

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26815

Differential Revision: [D17575905](https://our.internmc.facebook.com/intern/diff/D17575905)

zou3519 committed:
Named tensor support for: all, any, bitwise_not, cumprod, cumsum, and more

This PR adds named tensor support for:
- any, all, `bitwise_not(_)`, cumprod, cumsum, `type_as`, logical_not

In addition, it adds smoke tests for a variety of tensor attributes and
fns:
- is_shared, is_signed
- retain_grad, register_hook

Test Plan:
- [namedtensor ci]

nairbv commented on test/test_namedtensor.py:
```diff
@@ -544,6 +555,13 @@ def test_split_fns_propagates_names(self):
+    def test_any_all(self):
+        for device in torch.testing.get_all_device_types():
+            x = torch.zeros(3, dtype=torch.bool, device=device, names=('C',))
+            # The result can't have any names so this is just a smoke test
```
should we explicitly assert that dim names is empty?

zou3519 replied:
It's impossible for a scalar tensor to have any names: there is an invariant that a tensor must have number of names equal to number of dimensions. I think asserting it here would be redundant because other tests also make this assumption.

nairbv replied:
right, but we could test where we reduce a multi-dim tensor along a single dimension.

zou3519 replied:
That's a good point, those tests exist. I've tossed the check in that the names are empty.

zou3519 committed:
Update on "Named tensor support for: all, any, bitwise_not, cumprod, cumsum, and more"


This PR adds named tensor support for:
- any, all, `bitwise_not(_)`, cumprod, cumsum, `logical_not`

In addition, it adds smoke tests for a variety of tensor attributes and
fns:
- is_shared, is_signed
- retain_grad, register_hook

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26815

zou3519 committed:
Update on "Named tensor support for: all, any, bitwise_not, cumprod, cumsum, and more"


This PR adds named tensor support for:
- any, all, `bitwise_not(_)`, cumprod, cumsum, `logical_not`

In addition, it adds smoke tests for a variety of tensor attributes and
fns:
- is_shared, is_signed
- retain_grad, register_hook

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26815

Differential Revision: [D17575905](https://our.internmc.facebook.com/intern/diff/D17575905)

zou3519 committed:
Update on "Named tensor support for: all, any, bitwise_not, cumprod, cumsum, and more"


This PR adds named tensor support for:
- any, all, `bitwise_not(_)`, cumprod, cumsum, `logical_not`

In addition, it adds smoke tests for a variety of tensor attributes and
fns:
- is_shared, is_signed
- retain_grad, register_hook

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26815

Differential Revision: [D17575905](https://our.internmc.facebook.com/intern/diff/D17575905)
